patches/server/0065-Highly-boost-chunk-lighting-behavior.patch

```diff
@@ -0,0 +1,95 @@
+index 4d024a4849b7566c0faba131caab57b4ddf35e29..1f587f007074363c0f1e0bd9d6d83184eb039ab4 100644
+--- a/src/main/java/net/minecraft/server/LightEngineThreaded.java
++++ b/src/main/java/net/minecraft/server/LightEngineThreaded.java
+@@ -264,6 +264,7 @@ public class LightEngineThreaded extends LightEngine implements AutoCloseable {
+     }
+ 
+     public CompletableFuture<IChunkAccess> a(IChunkAccess ichunkaccess, boolean flag) {
++
+         ChunkCoordIntPair chunkcoordintpair = ichunkaccess.getPos();
+ 
+         // Paper start
+@@ -272,6 +273,13 @@ public class LightEngineThreaded extends LightEngine implements AutoCloseable {
+         CompletableFuture<IChunkAccess> future = new CompletableFuture<>();
+         IntSupplier prioritySupplier = playerChunkMap.getPrioritySupplier(pair);
+         boolean[] skippedPre = {false};
++        boolean[] asyncPre = {net.yatopia.server.YatopiaConfig.asyncLighting}; // Yatopia
++        // Yatopia start - unblock chunk loading
++        if(asyncPre[0]) {
++            future.complete(ichunkaccess);
++            this.d.c(chunkcoordintpair);
++        }
++        // Yatopia end
+         this.queue.addChunk(pair, prioritySupplier, SystemUtils.a(() -> {
+             if (!isChunkLightStatus(pair)) {
+                 future.complete(ichunkaccess);
+@@ -301,12 +309,12 @@ public class LightEngineThreaded extends LightEngine implements AutoCloseable {
+             return "lightChunk " + chunkcoordintpair + " " + flag;
+             // Paper start  - merge the 2 together
+         }), () -> {
+-            this.d.c(chunkcoordintpair); // Paper - release light tickets as post task to ensure they stay loaded until fully done
++            if(!asyncPre[0]) this.d.c(chunkcoordintpair); // Paper - release light tickets as post task to ensure they stay loaded until fully done // Yatopia - don't double release ticket
+             if (skippedPre[0]) return; // Paper - future's already complete
+             ichunkaccess.b(true);
+             super.b(chunkcoordintpair, false);
+             // Paper start
+-            future.complete(ichunkaccess);
++            if(!asyncPre[0]) future.complete(ichunkaccess); // Yatopia - don't double complete future
+         });
+         return future;
+         // Paper end
+diff --git a/src/main/java/net/yatopia/server/YatopiaConfig.java b/src/main/java/net/yatopia/server/YatopiaConfig.java
+index f0e7684304cab9cea05fa266a66a2ad8172c55b6..b63c880c54d97487ae183a27d761173b7af4f157 100644
+--- a/src/main/java/net/yatopia/server/YatopiaConfig.java
++++ b/src/main/java/net/yatopia/server/YatopiaConfig.java
+@@ -253,4 +253,9 @@ public class YatopiaConfig {
+         checkFlying = getBoolean("settings.checks.flight", checkFlying);
+         checkVehicleFlying = getBoolean("settings.checks.vehicle-flight", checkVehicleFlying);
+     }
++
++    public static boolean asyncLighting = true;
```

Comment:
Remove that option. It will be like paper's "async chunks" option, which got misunderstood by people, and they disabled it, and then suffered from performance issues.